packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -217,7 +218,7 @@ abstract class Route<T> {
   TickerFuture didPush() {
     return TickerFuture.complete()..then<void>((void _) {
       if (navigator?.widget.requestFocus ?? false) {
-        navigator!.focusScopeNode.requestFocus();
+        navigator!.focusNode.enclosingScope?.requestFocus();
```

Comment:
No, because if a nested navigator pops a route and comes back, then requesting focus on the enclosing scope will refocus the previously focused item in the scope.